ci: use the release access token on checkout

jacderida opened the pull request:
This may be what's causing the failure to push to the repository. The version bumping workflow uses a PAT on the checkout action, so hopefully this may resolve the issue.

## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 08 Jun 23 22:54 UTC
This pull request fixes an issue with the push to the repository during the version bumping workflow. The release access token is now used during checkout to resolve the problem.
<!-- reviewpad:summarize:end -->

jacderida commented on .github/workflows/pr_tests.yml:
```diff
@@ -105,3 +81,135 @@ jobs:
+        run: rm -rf ~/.safe/node
+
+      - name: Run example of split and chunk check
+        if: false
```
Should we include these 2 steps if they're not going to run? Or is that because they might be re-enabled in the future?

oetyng replied:
Yeah, good point. I'm not sure if we prefer commenting out, keeping or deleting.
But just running 15 nodes I think is very temporary, until we get splits stabilizing, and so I've followed that, except here. To be consistent we should comment these out as well.

oetyng replied:
..oor IDK, this is separate. Why do you mean they are not going to run?

jacderida replied:
The condition is set to `false`, so they'll never run as they are just now.

joshuef replied:
this wont run atm due to `false`. 

I think this should be able to be re-enabled (just remove the `if`)
```suggestion

```

jacderida commented on .github/workflows/pr_tests.yml:
```diff
@@ -105,3 +81,135 @@ jobs:
+        env:
+          NODE_COUNT: 15 # TODO: change it back to 43 (this should be a split section)
+
+      # - name: extra wait for CI split
```
Btw, in general I'm not a fan of leaving commented out code in commits. Do we want to keep this around for some kind of reference?

joshuef replied:
naw we dont need that if it's not being used atm (that's a hangover from the e2e file i believe). kill it

oetyng replied:
These numbers (node count, i.e. split or not split in e2e) keep changing back and forth all the time while we are not stable.

I guess it is too much of a hassle to delete and lookup/write again while we're not stable.

So, yeah, all the commented out stuff are things that would be brought back when NODE_COUNT is set to 43 again.

jacderida replied:
Cool, it's not a huge big deal either way.

jacderida committed:
ci: use the release access token on checkout

This may be what's causing the failure to push to the repository. The version bumping workflow uses
a PAT on the checkout action, so hopefully this may resolve the issue.
